packages/server/src/api/routes/utils/validators.ts

```diff
@@ -20,29 +20,28 @@ const OPTIONAL_NUMBER = Joi.number().optional().allow(null)
 const OPTIONAL_BOOLEAN = Joi.boolean().optional().allow(null)
 const APP_NAME_REGEX = /^[\w\s]+$/
 
-const validateViewSchemas: CustomValidator<Table> = (table, helpers) => {
-  if (table.views && Object.entries(table.views).length) {
-    const requiredFields = Object.entries(table.schema)
-      .filter(([_, v]) => isRequired(v.constraints))
+const validateViewSchemas: CustomValidator<Table> = (table, joiHelpers) => {
+  if (!table.views || Object.keys(table.views).length === 0) {
+    return table
+  }
+  const required = Object.keys(table.schema).filter(key =>
+    isRequired(table.schema[key].constraints)
+  )
+  if (required.length === 0) {
+    return table
+  }
+  for (const view of Object.values(table.views)) {
+    if (!sdk.views.isV2(view) || helpers.views.isCalculationView(view)) {
+      continue
+    }
```

Comment:
This is the only meaningful change in here, but I did a little refactoring to reduce the nesting and make this more readable while I was here.